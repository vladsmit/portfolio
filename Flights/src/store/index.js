import { createStore, applyMiddleware, compose } from 'redux';
import { createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';
import thunk from 'redux-thunk';

import RootReducers from './reducers';
import middlewares from './middlewares';

export const history = createBrowserHistory();

const composer = window.__REDUX_DEVTOOLS_EXTENSION__ ? compose(applyMiddleware(routerMiddleware(history), thunk, ...middlewares), window.__REDUX_DEVTOOLS_EXTENSION__(),) : compose(applyMiddleware(routerMiddleware(history), thunk, ...middlewares),);

export const store = createStore(RootReducers(history), composer,);