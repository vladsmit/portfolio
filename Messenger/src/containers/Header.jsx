import React, { useCallback } from 'react';
import { push } from 'connected-react-router';
import PropTypes from 'prop-types';
import connect from 'react-redux/es/connect/connect';
import { bindActionCreators } from 'redux';

const Header = (props) => {
    const handleNavigate = useCallback(link => {
        props.push(link);
    }, []);

    return (
        <header className="mainHeader">
            <a className="mainHeader__profileLink" onClick={() => handleNavigate('/profile')}>
                <img src="/dist/build/img/profile.png" alt="profileLogo" width="40"/>{props.profile.name}
            </a>
            <span className="mainHeader__chatName mainHeader__chatName--margin">{props.chats[props.chatId].title}</span>
        </header>
    );
};

Header.propTypes = {
    profile: PropTypes.object.isRequired,
    chats: PropTypes.object.isRequired,
    chatId: PropTypes.number.isRequired,
    push: PropTypes.func.isRequired,
};

const mapStateToProps = ({ chatReducer }) => ({
    profile: chatReducer.profile,
    chats: chatReducer.chats,
});

const mapDispatchToProps = dispatch => bindActionCreators({ push }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Header);