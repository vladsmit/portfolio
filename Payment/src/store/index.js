import Vue from 'vue';
import Vuex from 'vuex';
import createPersistedState from 'vuex-persistedstate';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    validationPassed: false,
    accounts: []
  },
  
  getters: {
    getValidationStatus(state) {
      return state.validationPassed;
    },

    getAccounts(state) {
      return state.accounts;
    },

    getAccountsLength(state, getters) {
      return getters.getAccounts.length;
    }
  },
  
  mutations: {
    changeValidationPassed(state) {
      state.validationPassed = !state.validationPassed;
    },

    addPayment(state, payload) {
      state.accounts = [...state.accounts, {
        id: state.accounts.length + 1,
        accountNumber: payload.accountNumber,
        sum: payload.sum,
        cardNumber: payload.cardNumber,
        owner: payload.owner,
        date: payload.date
      }];
    }
  },
  
  plugins: [createPersistedState()]
});
