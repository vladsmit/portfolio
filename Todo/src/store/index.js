import Vue from 'vue';
import Vuex from 'vuex';
import Axios from 'axios';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    taskList: [],
    page: null,
  },

  getters: {
    getTaskList(state) {
      return state.taskList;
    },

    getPage(state) {
      return state.page;
    },

    paginationPageList(state) {
      let start = (state.page - 1) * 10;
      let end = state.page * 10;
      return state.taskList.slice(start, end);
    }
  },

  mutations: {
    saveInLocalStorage(state, payload) {
      localStorage.setItem('taskList', JSON.stringify(payload));
    },

    setTaskList(state) {
      if (localStorage.taskList) {
        let taskList = JSON.parse(localStorage.taskList);
        state.taskList = taskList;
      }
    },

    reverseTaskList(state, payload) {
      payload.reverse();
    },

    addTaskInTaskList(state, payload) {
      state.taskList.unshift({
        id: payload.id,
        title: payload.title,
        completed: payload.completed,
        edit: payload.edit
      });
    },

    deleteTaskFromTaskList(state, payload) {
      state.taskList = state.taskList.filter((task) => payload !== task);
    },

    incrementPage(state) {
      state.page++;
    },

    decrementPage(state) {
      state.page--;
    },

    pageAmount(state, payload) {
      state.page = payload.amount;
    }
  },

  actions: {
    async getTasksFromJson({ commit }) {
      let { data } = await Axios.get("../tasks.json");
      commit('reverseTaskList', data);
      commit('saveInLocalStorage', data);
      commit('setTaskList');
    },

    addTaskInLocalStorage({ commit, state }, payload) {
      commit('addTaskInTaskList', payload);
      commit('saveInLocalStorage', state.taskList);
    },

    deleteTaskFromLocalStorage({ commit, state }, payload) {
      commit('deleteTaskFromTaskList', payload);
      commit('saveInLocalStorage', state.taskList);
    },
  }
});