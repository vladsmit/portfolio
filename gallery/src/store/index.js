import Vue from 'vue'
import Vuex from 'vuex'
import Axios from 'axios'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    categories: null,
    loading: false
  },

  getters: {
    getLoadingStatus(state) {
      return state.loading;
    },

    getCategories(state) {
      return state.categories;
    }
  },

  mutations: {
    isLoading(state, payload) {
      state.loading = payload;
    },

    savePhotosInState(state, payload) {
      let photos = [];
      let num = 0;
      for (let i = 0; i < 4; i++) {
        photos.push([]);
        for (let j = num; j < 24; j++) {
          if (photos[i].length < 6) {
            photos[i].push(payload[j]);
          } else {
            break;
          }
        }
        num += 6;
      }
      state.categories = photos;
    },
  },

  actions: {
    async getPhotosFromApi({ commit }) {
      commit('isLoading', true);
      const { data } = await Axios.get("https://jsonplaceholder.typicode.com/photos");
      setTimeout(() => {
        commit('savePhotosInState', data);
        commit('isLoading', false);
      }, 500);
    }
  },
})
