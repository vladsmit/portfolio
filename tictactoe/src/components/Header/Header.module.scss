.text,
.playerName {
  font-size: 28px;
  font-weight: 700;
  text-align: center;
  color: #0060e6;
}

.gameCountBlock {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;

  .playerInfoBlock {
    width: 35%;
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .playerName {
    width: 100%;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
  }
}

.buttonFontWeight {
  font-weight: 600;
}
