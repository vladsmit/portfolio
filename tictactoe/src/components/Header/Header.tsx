import { Button } from 'antd';

import { useAppSelector } from '../../hooks';
import { selectPlayers } from '../../store/slices/ticTacToeSlice';
import styles from './Header.module.scss';

interface IHeaderProps {
  setIsOpen: () => void;
}

export const Header = (props: IHeaderProps) => {
  const { setIsOpen } = props;

  const { cross: firstPlayer, zero: secondPlayer } =
    useAppSelector(selectPlayers);

  return (
    <>
      <h1 className={styles.text}>Крестики-нолики</h1>
      <div className={styles.gameCountBlock}>
        <div className={styles.playerInfoBlock}>
          <span className={styles.playerName}>{firstPlayer.name}</span>
          <span className={styles.text}>{firstPlayer.wins}</span>
        </div>

        <Button
          className={styles.buttonFontWeight}
          type="text"
          onClick={setIsOpen}
        >
          Новая серия
        </Button>

        <div className={styles.playerInfoBlock}>
          <span className={styles.playerName}>{secondPlayer.name}</span>
          <span className={styles.text}>{secondPlayer.wins}</span>
        </div>
      </div>
    </>
  );
};
