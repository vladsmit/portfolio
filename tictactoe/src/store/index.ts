import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import ticTacToeReducer from './slices/ticTacToeSlice';

export const store = configureStore({
  reducer: {
    ticTacToe: ticTacToeReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
