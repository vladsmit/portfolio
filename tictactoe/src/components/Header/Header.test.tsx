import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import { store } from '../../store';
import { Header } from './Header';

describe('Header component', () => {
  it('should render Header component', () => {
    const onClick = jest.fn();
    
    const component = render(
      <Provider store={store}>
        <Header setIsOpen={onClick} />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });
});
