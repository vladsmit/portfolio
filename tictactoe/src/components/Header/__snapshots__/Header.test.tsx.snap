// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text"
      >
        Крестики-нолики
      </h1>
      <div
        class="gameCountBlock"
      >
        <div
          class="playerInfoBlock"
        >
          <span
            class="playerName"
          >
            Игрок 1
          </span>
          <span
            class="text"
          >
            0
          </span>
        </div>
        <button
          class="ant-btn css-dev-only-do-not-override-k7429z ant-btn-text buttonFontWeight"
          type="button"
        >
          <span>
            Новая серия
          </span>
        </button>
        <div
          class="playerInfoBlock"
        >
          <span
            class="playerName"
          >
            Игрок 2
          </span>
          <span
            class="text"
          >
            0
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1
      class="text"
    >
      Крестики-нолики
    </h1>
    <div
      class="gameCountBlock"
    >
      <div
        class="playerInfoBlock"
      >
        <span
          class="playerName"
        >
          Игрок 1
        </span>
        <span
          class="text"
        >
          0
        </span>
      </div>
      <button
        class="ant-btn css-dev-only-do-not-override-k7429z ant-btn-text buttonFontWeight"
        type="button"
      >
        <span>
          Новая серия
        </span>
      </button>
      <div
        class="playerInfoBlock"
      >
        <span
          class="playerName"
        >
          Игрок 2
        </span>
        <span
          class="text"
        >
          0
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
